### PR TITLE
Vegetarian/Vegan markup

### PR DESCRIPTION
**Pull request by aleene**
Started marking up vegetarian and vegan with:

vegetarian:en:yes
vegan:en:yes

It assumed that vegetarian is a superset of vegan.

### CODE REVIEW

**Comment by stephanegigandet** [comment]
Can you add vegetarian:en:yes even if there is already vegan:en:yes ? It is going to make things much simpler on the backend, the app, and third party apps if they don't have to hardcode special processing.

Also the negative values are very important as well, so if something is vegetarian:en:yes but not vegan, we need the vegan:en:no

**Comment by aleene** [comment]
Ok, i'll do the vegetarian=no as well. So we can see nil values as well (those that have no been look at, or no info).

**Comment by teolemon** [comment]
@aleene merge conflict to solve

**Comment by aleene** [comment]
I do not see any conflicts in my file, so what is it? test fails?

File: taxonomies/ingredients.txt [hunk 1]
```diff
@@ -61062,6 +60950,8 @@ zu:Ikhofi
 # zh-tw:咖啡
 wikidata:en:Q8486
 wikipedia:en:https://en.wikipedia.org/wiki/Coffee
+vegan:en:yes
+vegetarian:en:yes
 # ingredient/en:coffee has 1637 products in 10 languages @2018-11-30
```
Comment (teolemon) [inline taxonomies/ingredients.txt]:
```suggestion
# ingredient/en:coffee has 1637 products in 10 languages @2018-11-30 -
```

**Comment by VaiTon** [comment]
@aleene there are still conflicts

**Comment by teolemon** [comment]
```  
#   Failed test at t/additives_tags.t line 38.
#     Structures begin differing at:
#          $got->[0] = 'en:e120'
#     $expected->[0] = Does not exist
# {
#   'additives' => ' [ laitue -> fr:laitue  ]  [ carmine -> en:e120  -> exists  -- ok  ] ',
#   'additives_n' => 1,
#   'additives_old_n' => 1,
#   'additives_old_tags' => [
#     'en:e120'
#   ],
#   'additives_original_tags' => [
#     'en:e120'
#   ],
#   'additives_tags' => [
#     'en:e120'
#   ],
#   'amino_acids_tags' => [],
#   'categories_tags' => [
#     'en:debug'
#   ],
#   'ingredients_from_or_that_may_be_from_palm_oil_n' => 0,
#   'ingredients_from_palm_oil_n' => 0,
#   'ingredients_from_palm_oil_tags' => [],
#   'ingredients_text' => 'Laitue, Carmine',
#   'ingredients_text_fr' => 'Laitue, Carmine',
#   'ingredients_that_may_be_from_palm_oil_n' => 0,
#   'ingredients_that_may_be_from_palm_oil_tags' => [],
#   'lc' => 'fr',
#   'minerals_tags' => [],
#   'nucleotides_tags' => [],
#   'other_nutritional_substances_tags' => [],
#   'vitamins_tags' => []
# }
```

**Comment by teolemon** [comment]
``` 
#   Failed test at t/ingredients.t line 708.
#     Structures begin differing at:
#          $got->{ingredients_hierarchy}[22] = 'en:flour'
#     $expected->{ingredients_hierarchy}[22] = 'en:cereal-flour'
# {
#   'ingredients' => [
#     {
#       'id' => 'fr:Marmelade d\'oranges',
#       'percent' => '41',
#       'rank' => 1,
#       'text' => 'Marmelade d\'oranges'
#     },
#     {
#       'id' => "fr:pulpe d'orange concentr\x{e9}e",
#       'percent' => '0.6',
#       'rank' => 2,
#       'text' => "pulpe d'orange concentr\x{e9}e"
#     },
#     {
#       'id' => "fr:\x{e9}quivalent pulpe d'orange",
#       'percent' => '2.6',
#       'rank' => 3,
#       'text' => "\x{e9}quivalent pulpe d'orange"
#     },
#     {
#       'id' => 'en:gelling-agent',
#       'rank' => 4,
#       'text' => "g\x{e9}lifiant"
#     },
#     {
#       'id' => 'en:e440a',
#       'rank' => 5,
#       'text' => 'pectines'
#     },
#     {
#       'id' => 'en:acid',
#       'rank' => 6,
#       'text' => 'acidifiant'
#     },
#     {
#       'id' => 'en:e330',
#       'rank' => 7,
#       'text' => 'acide citrique'
#     },
#     {
#       'id' => 'en:acidity-regulator',
#       'rank' => 8,
#       'text' => "correcteurs d'acidit\x{e9}"
#     },
#     {
#       'id' => 'en:natural-orange-flavouring',
#       'rank' => 9,
#       'text' => "ar\x{f4}me naturel d'orange"
#     },
#     {
#       'id' => 'en:thickener',
#       'rank' => 10,
#       'text' => "\x{e9}paississant"
#     },
#     {
#       'id' => 'en:e415',
#       'rank' => 11,
#       'text' => 'gomme xanthane'
#     },
#     {
#       'id' => 'en:chocolate',
#       'percent' => '24.9',
#       'rank' => 12,
#       'text' => 'chocolat'
#     },
#     {
#       'id' => 'en:lactose-and-milk-proteins',
#       'rank' => 13,
#       'text' => "lactose et prot\x{e9}ines de lait"
#     },
#     {
#       'id' => 'en:wheat-flour',
#       'rank' => 14,
#       'text' => "farine de bl\x{e9}"
#     },
#     {
#       'id' => 'en:sugar',
#       'rank' => 15,
#       'text' => 'sucre'
#     },
#     {
#       'id' => 'en:egg',
#       'rank' => 16,
#       'text' => 'oeufs'
#     },
#     {
#       'id' => 'en:glucose-fructose-syrup',
#       'rank' => 17,
#       'text' => 'sirop de glucose-fructose'
#     },
#     {
#       'id' => 'en:colza-oil',
#       'rank' => 18,
#       'text' => 'huile de colza'
#     },
#     {
#       'id' => 'en:raising-agent',
#       'rank' => 19,
#       'text' => "poudre \x{e0} lever"
#     },
#     {
#       'id' => 'en:salt',
#       'rank' => 20,
#       'text' => 'sel'
#     },
#     {
#       'id' => 'en:emulsifier',
#       'rank' => 21,
#       'text' => "\x{e9}mulsifiant"
#     },
#     {
#       'id' => 'en:soya-lecithin',
#       'rank' => 22,
#       'text' => "l\x{e9}cithine de soja"
#     },
#     {
#       'id' => 'en:glucose-fructose-syrup',
#       'text' => 'sirop de glucose-fructose'
#     },
#     {
#       'id' => 'en:sugar',
#       'text' => 'sucre'
#     },
#     {
#       'id' => 'en:orange-pulp',
#       'percent' => '4.5',
#       'text' => 'pulpe d\'orange'
#     },
#     {
#       'id' => 'en:concentrated-orange-juice',
#       'percent' => '1.4',
#       'text' => "jus d'orange concentr\x{e9}"
#     },
#     {
#       'id' => "fr:\x{e9}quivalent jus d'orange",
#       'percent' => '7.8',
#       'text' => "\x{e9}quivalent jus d'orange"
#     },
#     {
#       'id' => 'en:e333',
#       'text' => 'citrate de calcium'
#     },
#     {
#       'id' => 'en:sodium-citrate',
#       'text' => 'citrate de sodium'
#     },
#     {
#       'id' => 'en:sugar',
#       'text' => 'sucre'
#     },
#     {
#       'id' => 'en:cocoa-paste',
#       'text' => "p\x{e2}te de cacao"
#     },
#     {
#       'id' => 'en:cocoa-butter',
#       'text' => 'beurre de cacao'
#     },
#     {
#       'id' => "fr:graisses v\x{e9}g\x{e9}tales d'illipe",
#       'text' => "graisses v\x{e9}g\x{e9}tales d'illipe"
#     },
#     {
#       'id' => "fr:graisses v\x{e9}g\x{e9}tales de mangue",
#       'text' => "graisses v\x{e9}g\x{e9}tales de mangue"
#     },
#     {
#       'id' => "fr:graisses v\x{e9}g\x{e9}tales de sal",
#       'text' => "graisses v\x{e9}g\x{e9}tales de sal"
#     },
#     {
#       'id' => 'en:shea-butter',
#       'text' => "graisses v\x{e9}g\x{e9}tales de karit\x{e9}"
#     },
#     {
#       'id' => 'en:palm-fat',
#       'text' => "graisses v\x{e9}g\x{e9}tales de palme"
#     },
#     {
#       'id' => 'en:flavouring',
#       'text' => "ar\x{f4}me"
#     },
#     {
#       'id' => 'en:emulsifier',
#       'text' => "\x{e9}mulsifiant"
#     },
#     {
#       'id' => 'en:soya-lecithin',
#       'text' => "l\x{e9}cithine de soja"
#     },
#     {
#       'id' => 'en:e503',
#       'text' => 'carbonate acide d\'ammonium'
#     },
#     {
#       'id' => 'en:e450i',
#       'text' => 'diphosphate disodique'
#     },
#     {
#       'id' => 'en:e500',
#       'text' => 'carbonate acide de sodium'
#     }
#   ],
#   'ingredients_hierarchy' => [
#     'fr:Marmelade d\'oranges',
#     "fr:pulpe d'orange concentr\x{e9}e",
#     "fr:\x{e9}quivalent pulpe d'orange",
#     'en:gelling-agent',
#     'en:e440a',
#     'en:acid',
#     'en:e330',
#     'en:acidity-regulator',
#     'en:natural-orange-flavouring',
#     'en:flavouring',
#     'en:natural-flavouring',
#     'en:thickener',
#     'en:e415',
#     'en:chocolate',
#     'en:lactose-and-milk-proteins',
#     'en:protein',
#     'en:animal-protein',
#     'en:milk-proteins',
#     'en:lactose',
#     'en:wheat-flour',
#     'en:cereal',
#     'en:wheat',
#     'en:flour',
#     'en:cereal-flour',
#     'en:sugar',
#     'en:egg',
#     'en:glucose-fructose-syrup',
#     'en:glucose',
#     'en:fructose',
#     'en:colza-oil',
#     'en:oil',
#     'en:vegetable-oil-and-fat',
#     'en:vegetable-oil',
#     'en:raising-agent',
#     'en:salt',
#     'en:emulsifier',
#     'en:soya-lecithin',
#     'en:e322',
#     'en:orange-pulp',
#     'en:fruit',
#     'en:citrus-fruit',
#     'en:orange',
#     'en:concentrated-orange-juice',
#     'en:fruit-juice',
#     'en:orange-juice',
#     "fr:\x{e9}quivalent jus d'orange",
#     'en:e333',
#     'en:sodium-citrate',
#     'en:minerals',
#     'en:sodium',
#     'en:cocoa-paste',
#     'en:cocoa',
#     'en:cocoa-butter',
#     "fr:graisses v\x{e9}g\x{e9}tales d'illipe",
#     "fr:graisses v\x{e9}g\x{e9}tales de mangue",
#     "fr:graisses v\x{e9}g\x{e9}tales de sal",
#     'en:shea-butter',
#     'en:vegetable-fat',
#     'en:palm-fat',
#     'en:palm-oil-and-fat',
#     'en:e503',
#     'en:e450i',
#     'en:e450',
#     'en:e500'
#   ],
#   'ingredients_n' => 43,
#   'ingredients_n_tags' => [
#     '43',
#     '41-50'
#   ],
#   'ingredients_original_tags' => [
#     'fr:Marmelade d\'oranges',
#     "fr:pulpe d'orange concentr\x{e9}e",
#     "fr:\x{e9}quivalent pulpe d'orange",
#     'en:gelling-agent',
#     'en:e440a',
#     'en:acid',
#     'en:e330',
#     'en:acidity-regulator',
#     'en:natural-orange-flavouring',
#     'en:thickener',
#     'en:e415',
#     'en:chocolate',
#     'en:lactose-and-milk-proteins',
#     'en:wheat-flour',
#     'en:sugar',
#     'en:egg',
#     'en:glucose-fructose-syrup',
#     'en:colza-oil',
#     'en:raising-agent',
#     'en:salt',
#     'en:emulsifier',
#     'en:soya-lecithin',
#     'en:glucose-fructose-syrup',
#     'en:sugar',
#     'en:orange-pulp',
#     'en:concentrated-orange-juice',
#     "fr:\x{e9}quivalent jus d'orange",
#     'en:e333',
#     'en:sodium-citrate',
#     'en:sugar',
#     'en:cocoa-paste',
#     'en:cocoa-butter',
#     "fr:graisses v\x{e9}g\x{e9}tales d'illipe",
#     "fr:graisses v\x{e9}g\x{e9}tales de mangue",
#     "fr:graisses v\x{e9}g\x{e9}tales de sal",
#     'en:shea-butter',
#     'en:palm-fat',
#     'en:flavouring',
#     'en:emulsifier',
#     'en:soya-lecithin',
#     'en:e503',
#     'en:e450i',
#     'en:e500'
#   ],
#   'ingredients_tags' => [
#     'fr:marmelade-d-oranges',
#     'fr:pulpe-d-orange-concentree',
#     'fr:equivalent-pulpe-d-orange',
#     'en:gelling-agent',
#     'en:e440a',
#     'en:acid',
#     'en:e330',
#     'en:acidity-regulator',
#     'en:natural-orange-flavouring',
#     'en:flavouring',
#     'en:natural-flavouring',
#     'en:thickener',
#     'en:e415',
#     'en:chocolate',
#     'en:lactose-and-milk-proteins',
#     'en:protein',
#     'en:animal-protein',
#     'en:milk-proteins',
#     'en:lactose',
#     'en:wheat-flour',
#     'en:cereal',
#     'en:wheat',
#     'en:flour',
#     'en:cereal-flour',
#     'en:sugar',
#     'en:egg',
#     'en:glucose-fructose-syrup',
#     'en:glucose',
#     'en:fructose',
#     'en:colza-oil',
#     'en:oil',
#     'en:vegetable-oil-and-fat',
#     'en:vegetable-oil',
#     'en:raising-agent',
#     'en:salt',
#     'en:emulsifier',
#     'en:soya-lecithin',
#     'en:e322',
#     'en:orange-pulp',
#     'en:fruit',
#     'en:citrus-fruit',
#     'en:orange',
#     'en:concentrated-orange-juice',
#     'en:fruit-juice',
#     'en:orange-juice',
#     'fr:equivalent-jus-d-orange',
#     'en:e333',
#     'en:sodium-citrate',
#     'en:minerals',
#     'en:sodium',
#     'en:cocoa-paste',
#     'en:cocoa',
#     'en:cocoa-butter',
#     'fr:graisses-vegetales-d-illipe',
#     'fr:graisses-vegetales-de-mangue',
#     'fr:graisses-vegetales-de-sal',
#     'en:shea-butter',
#     'en:vegetable-fat',
#     'en:palm-fat',
#     'en:palm-oil-and-fat',
#     'en:e503',
#     'en:e450i',
#     'en:e450',
#     'en:e500'
#   ],
#   'ingredients_text' => "Marmelade d'oranges 41% (sirop de glucose-fructose, sucre, pulpe d'orange 4.5%, jus d'orange concentr\x{e9} 1.4% (\x{e9}quivalent jus d'orange 7.8%), pulpe d'orange concentr\x{e9}e 0.6% (\x{e9}quivalent pulpe d'orange 2.6%), g\x{e9}lifiant (pectines), acidifiant (acide citrique), correcteurs d'acidit\x{e9} (citrate de calcium, citrate de sodium), ar\x{f4}me naturel d'orange, \x{e9}paississant (gomme xanthane)), chocolat 24.9% (sucre, p\x{e2}te de cacao, beurre de cacao, graisses v\x{e9}g\x{e9}tales (illipe, mangue, sal, karit\x{e9} et palme en proportions variables), ar\x{f4}me, \x{e9}mulsifiant (l\x{e9}cithine de soja), lactose et prot\x{e9}ines de lait), farine de bl\x{e9}, sucre, oeufs, sirop de glucose-fructose, huile de colza, poudre \x{e0} lever (carbonate acide d'ammonium, diphosphate disodique, carbonate acide de sodium), sel, \x{e9}mulsifiant (l\x{e9}cithine de soja).",
#   'lc' => 'fr',
#   'unknown_ingredients_n' => 7
# }
```

**Comment by teolemon** [comment]
``` 
#   Failed test at t/additives.t line 82.
#          got: ' [ acide-citrique -> en:e330  -> exists  -- ok  ]  [ colorant -> fr:colorant  ]  [ e120 -> en:e120  -> exists  -- ok  ]  [ vitamine-c -> en:e300  -> exists  -- mandatory_additive_class: en:acidity-regulator,en:antioxidant,en:flour-treatment-agent,en:sequestrant,en:acid (current: en:colour)  -> exists as a vitamin en:vitamin-c  ]  [ e500 -> en:e500  -> exists  -- mandatory_additive_class: en:acidity-regulator, en:raising-agent (current: en:vitamins)  -- e-number  ] '
#     expected: ' [ acide-citrique -> en:e330  -> exists  -- ok  ]  [ colorant -> fr:colorant  ]  [ e120 -> en:e120  -> exists  -- mandatory_additive_class: en:colour (current: en:colour)  -- ok  ]  [ vitamine-c -> en:e300  -> exists  -- mandatory_additive_class: en:acidity-regulator,en:antioxidant,en:flour-treatment-agent,en:sequestrant,en:acid (current: en:colour)  -> exists as a vitamin en:vitamin-c  ]  [ e500 -> en:e500  -> exists  -- mandatory_additive_class: en:acidity-regulator, en:raising-agent (current: en:vitamins)  -- e-number  ] '
```

File: taxonomies/additives.txt [hunk 2]
```diff
@@ -465,7 +483,8 @@ efsa_evaluation_url:en: http://dx.doi.org/10.2903/j.efsa.2015.4288
 efsa_evaluation_date:en: 2015-11-18
 efsa_evaluation:en: Scientific Opinion on the re-evaluation of cochineal, carminic acid, carmines (E 120) as a food additive
 additives_classes:en: en:colour
-mandatory_additive_class:en: en:colour
```
Comment (stephanegigandet) [inline taxonomies/additives.txt]: I'll add this back, we need it to make sure that "carmine" (some kind of salad) does not trigger E120

**Pull request merged**